aspnetcore/fundamentals/file-providers.md

```diff
@@ -1,148 +1,261 @@
+::: moniker-end
 
-You can use [globbing patterns](#globbing-patterns) when specifying files to embed in the assembly. These patterns can be used to match one or more files.
+::: moniker range=">= aspnetcore-2.0"
+
+### ManifestEmbeddedFileProvider
+
+The [ManifestEmbeddedFileProvider](/dotnet/api/microsoft.extensions.fileproviders.manifestembeddedfileprovider) is used to access files embedded within assemblies. The `ManifestEmbeddedFileProvider` uses a manifest compiled into the assembly to reconstruct the original paths of the embedded files.
 
 > [!NOTE]
-> It's unlikely you would ever want to actually embed every .js file in your project in its assembly; the above sample is for demo purposes only.
+> The `ManifestEmbeddedFileProvider` is available in ASP.NET Core 2.1 or later. To access files embedded in assemblies in ASP.NET Core 2.0 or earlier, see the [ASP.NET Core 1.x version of this topic](xref:fundamentals/file-providers?view=aspnetcore-1.1).
+
+To generate a manifest of the embedded files, set the **&lt;GenerateEmbeddedFilesManifest&gt;** property to `true`. Specify the files to embed with [&lt;EmbeddedResource&gt;](/dotnet/core/tools/csproj#default-compilation-includes-in-net-core-projects):
+
+[!code-csharp[](file-providers/samples/2.x/FileProviderSample/FileProviderSample.csproj?highlight=5,13)]
 
-When creating an `EmbeddedFileProvider`, pass the assembly it will read to its constructor.
+Use [glob patterns](#glob-patterns) to specify one or more files to embed into the assembly.
+
+The sample app creates an `ManifestEmbeddedFileProvider` and passes the currently executing assembly to its constructor.
+
+*Startup.cs*:
 
 ```csharp
-var embeddedProvider = new EmbeddedFileProvider(Assembly.GetEntryAssembly());
+var manifestEmbeddedProvider = 
+    new ManifestEmbeddedFileProvider(Assembly.GetEntryAssembly());
 ```
 
-The snippet above demonstrates how to create an `EmbeddedFileProvider` with access to the currently executing assembly.
+Additional overloads allow you to:
 
-Updating the sample app to use an `EmbeddedFileProvider` results in the following output:
+* Specify a relative file path.
+* Scope files to a last modified date.
+* Name the embedded resource containing the embedded file manifest.
 
-![File provider sample application listing embedded files](file-providers/_static/embedded-directory-listing.png)
+| Overload | Description |
+| -------- | ----------- |
+| [ManifestEmbeddedFileProvider(Assembly, String)](/dotnet/api/microsoft.extensions.fileproviders.manifestembeddedfileprovider.-ctor?view=aspnetcore-2.1#Microsoft_Extensions_FileProviders_ManifestEmbeddedFileProvider__ctor_System_Reflection_Assembly_System_String_) | Accepts an optional `root` relative path parameter. Specify the `root` to scope calls to [GetDirectoryContents](/dotnet/api/microsoft.extensions.fileproviders.ifileprovider.getdirectorycontents) to those resources under the provided path. |
```

Comment:
I recommend removing the version from the API ref page links. This section displays for 2.0+. If someone's looking at 2.0 or 2.2, they may wonder why the 2.1 API ref page is displayed.

Reply:
> removing the version from the API ref page links.

It was just an oversight ... I always do it ... well ... *almost always* :smile: